staging/src/k8s.io/client-go/tools/record/event.go

```diff
@@ -489,7 +489,7 @@ func (recorder *recorderImpl) makeEvent(ref *v1.ObjectReference, annotations map
 	}
 	return &v1.Event{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:        fmt.Sprintf("%v.%x", ref.Name, t.UnixNano()),
+			Name:        util.GenerateEventName(ref.Name, t.UnixNano()),
```

Comment:
same with https://github.com/kubernetes/kubernetes/blob/d99d3f7eb734dc9284a90374fcc6d8cc32d1946d/staging/src/k8s.io/client-go/tools/record/events_cache.go#L52-L68

Reply:
I commented in https://github.com/kubernetes/kubernetes/pull/129790#discussion_r1936326934, creating the event name depending on the name itself leaves the code in a weird state that may also cause issues in the future as it can make it recursive if we are not careful, this approach is simpler and solves the problem

Reply:
those concatenation of strings does not guarantee the generate name is a valid name, see new approach